### PR TITLE
Add stack canaries to both the payload and the loader

### PR DESCRIPTION
**Pull request by MikeIsAStar**

### CODE REVIEW

File: common/StackCanary.h [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#pragma once
+
+#include "common/Types.h"
+
+asm void StackCanary_Init(void);
+asm u32 StackCanary_Canary(void);
```
Comment (stblr) [inline common/StackCanary.h:6]: Can the `asm` part be moved to the c file?

**Review by stblr — COMMENTED** [review]
The commit message mentions the payload, but I'm pretty sure they are also added to the loader.

File: common/StackCanary.h (hunk 1 above)
Comment (MikeIsAStar) [reply common/StackCanary.h]: Resolved in commit [7716b67](https://github.com/doubledashdeluxe/ddd/pull/66/commits/7716b677b06aafacbcc399c725836f8cdc6ee1dc).

**Comment by MikeIsAStar** [comment]
> The commit message mentions the payload, but I'm pretty sure they are also added to the loader.

You are absolutely correct ! I have updated the commit message to reflect this.

**Review by stblr — APPROVED** [review]

**Pull request merged**